File: KrishiSetu.Api/Models/Booking.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;


namespace KrishiSetu.Api.Models
{
    public class Booking
    {
        [Key]
        public int Id { get; set; }
        
        public int FarmerId { get; set; }
        public User Farmer { get; set; } = null!;
        
        public int? MachineryId { get; set; }
        public Machinery? Machinery { get; set; }
        
        public int? WorkerId { get; set; }
        public WorkerProfile? Worker { get; set; }
        
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        
        public decimal TotalAmount { get; set; }
        
        public string Status { get; set; } = "Pending"; // Pending, Confirmed, Completed, Cancelled
        
        public string PaymentStatus { get; set; } = "Pending"; // Pending, Paid

        public ICollection<Payment> Payments { get; set; } = new List<Payment>();
        public ICollection<Earning> Earnings { get; set; } = new List<Earning>();
        public Review? Review { get; set; }
    }
}

File: krishisetu-backend/src/main/java/com/krishisetu/backend/entity/Booking.java
package com.krishisetu.backend.entity;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDate;

@Entity
@Table(name = "bookings")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Booking {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "farmer_id", nullable = false)
    private User farmer;

    // Can be null if booking is for a worker
    @ManyToOne
    @JoinColumn(name = "machinery_id")
    private Machinery machinery;

    // Can be null if booking is for machinery
    @ManyToOne
    @JoinColumn(name = "worker_profile_id")
    private WorkerProfile workerProfile;

    private LocalDate startDate;
    private LocalDate endDate;

    // For workers, we might book by hours
    private Integer hours;

    private Double amount;

    private String status; // PENDING, CONFIRMED, CANCELLED

    private String razorpayOrderId;
    private String razorpayPaymentId;
    private String paymentStatus; // PENDING, PAID, FAILED

    @JsonProperty("farmerName")
    public String getFarmerName() {
        return farmer != null ? farmer.getFirstName() + " " + farmer.getLastName() : "N/A";
    }

    @JsonProperty("totalAmount")
    public Double getTotalAmount() {
        return amount;
    }
}
